# Numerical tables for (insert name of paper here)

This notebook provides a searchable display for the tables of numerical values found in (paper). The first code cell below loads some Python packages needed to process the data.

In [1]:
import numpy as np
import pandas as pd
from pandas import DataFrame
import csv
from ipywidgets import widgets
import IPython

The next cell reads in the data from a collection CSV files generated (put information about how, who, etc. here).

The file names entered below will generate tables populated with $\LaTeX$-formatted numerical data. This will look nice, but you won't be able to work with the numbers. If you prefer to load numeric data (without formatting), append `_numeric` to the filename. (For example, replace `table9.csv` with `table9_numeric.csv`.)

In [2]:
table9 = pd.read_csv("table9.csv")
table10 = pd.read_csv("table10.csv",names=['$b_j$','$B_{j,1}$','$B_{j,2}$','$B_{j,3}$','$B_{j,4}$','$B_{j,5}$'])
table11 = pd.read_csv("table11.csv",names=['$b_j$','$B_{j,1}$','$B_{j,2}$','$B_{j,3}$','$B_{j,4}$','$B_{j,5}$'])
table13 = pd.read_csv("table13.csv",names=['$a$','$b$','$\mathcal{D}_0(a,b)$','$\mathcal{D}_1(a,b)$','$\mathcal{D}_2(a,b)$','$\mathcal{D}_3(a,b)$','$\mathcal{D}_4(a,b)$','$\mathcal{D}_5(a,b)$',])
table14 = pd.read_csv("table14.csv")
table15 = pd.read_csv("table15.csv")

### Table 9: Sharper bounds for $\psi(x)$ - $|\psi(x) - x| < \varepsilon(b) x$ where $\varepsilon(b)$ is computable for $x \ge e^b$ fixed

Combined table of sharpest bounds for $|\psi(x)-x|<\varepsilon x$. Computed using Wedeniwski's partial verification of the Riemann Hypothesis<br> ($H_0=2\,445\,999\,556\,030$ `[51]`). For all $x\ge x_0=e^{b}$ we have $|\psi(x)-x|<\varepsilon(b) x$. 

In Table 9, values for $b=20,\ldots, 43$ are computed as in `[7, Theorem 2]` and `[6, Theorem 1]`. Values for $b=19\log 10,  \ldots, 2300$ are computed as in `[6, Theorem 1]`. Values for $b=2325,\ldots, 13900$ are computed as in Theorem 3.

Here is the complete table:

In [9]:
table9.style.set_properties(**{'width': '120px'})

If you want to access particular parts of the table, here are some basic commands for the Pandas Python package:

- Type `table9.head()` to access the first few lines, or `table9.tail()` to get the last few.
- For a random sample of 5 lines, try `table9.sample(5)`. Of course, you can replace 5 by another value.

In [10]:
table9.head()

$b$             $\varepsilon(b)$
0  $20$   $ 4.26760 \cdot 10^{-5} $ 
1  $21$   $ 2.58843 \cdot 10^{-5} $ 
2  $22$   $ 1.56996 \cdot 10^{-5} $ 
3  $23$   $ 9.52229 \cdot 10^{-6} $ 
4  $24$   $ 5.77556 \cdot 10^{-6} $

In [11]:
table9[table9.$b$ == "29"]

SyntaxError: invalid syntax (<ipython-input-11-c0d0efc0930d>, line 1)

### Table 10: Sharper bounds for $\theta(x)$: $x$ in a middle range ($e^{20} \le x \le e^{13900}$) 

Values for $ B_{j,k}$ in $|\theta(x)-x|<\frac{ B_{j,k}x}{(\log x)^k}$ calculated using the method described in Corollary 14.1, 
Section 4.2.

Each $ B_{j,k}$ is 
valid for $e^{b_j}\le x \le e^{b_{j+1}}$. The last line is valid for $e^{10000} \le x \le e^{13900}$.

In [5]:
table10

$b_j$                   $B_{j,1}$                   $B_{j,2}$  \
0           $20 $     $1.8504 \cdot 10^{-3} $     $3.7905 \cdot 10^{-2} $    
1           $21 $     $1.1717 \cdot 10^{-3} $     $2.5175 \cdot 10^{-2} $    
2           $22 $     $7.4097 \cdot 10^{-4} $     $1.6663 \cdot 10^{-2} $    
3           $23 $     $4.6801 \cdot 10^{-4} $     $1.0993 \cdot 10^{-2} $    
4           $24 $     $2.9523 \cdot 10^{-4} $     $7.2298 \cdot 10^{-3} $    
5           $25 $     $1.8601 \cdot 10^{-4} $     $4.7413 \cdot 10^{-3} $    
6           $26 $     $1.1706 \cdot 10^{-4} $     $3.1009 \cdot 10^{-3} $    
7           $27 $     $7.3582 \cdot 10^{-5} $     $2.0228 \cdot 10^{-3} $    
8           $28 $     $4.6203 \cdot 10^{-5} $     $1.3164 \cdot 10^{-3} $    
9           $29 $     $2.8981 \cdot 10^{-5} $     $8.5467 \cdot 10^{-4} $    
10          $30 $     $1.8161 \cdot 10^{-5} $     $5.5373 \cdot 10^{-4} $    
11          $31 $     $1.1370 \cdot 10^{-5} $     $3.5803 \cdot 10^{-4} $    
12          $32 $     $7.1111 \cdot 10^{-6} $     $2.3105 \cdot 10^{-4} $    
13          $33 $     $4.4440 \cdot 10^{-6} $     $1.4884 \cdot 10^{-4} $    
14          $34 $     $2.7749 \cdot 10^{-6} $     $9.5708 \cdot 10^{-5} $    
15          $35 $     $1.7314 \cdot 10^{-6} $     $6.1447 \cdot 10^{-5} $    
16          $36 $     $1.2653 \cdot 10^{-6} $     $4.6264 \cdot 10^{-5} $    
17          $37 $     $1.0774 \cdot 10^{-6} $     $4.0598 \cdot 10^{-5} $    
18          $38 $     $9.6748 \cdot 10^{-7} $     $3.7519 \cdot 10^{-5} $    
19          $39 $     $9.0625 \cdot 10^{-7} $     $3.6118 \cdot 10^{-5} $    
20          $40 $     $8.7541 \cdot 10^{-7} $     $3.5809 \cdot 10^{-5} $    
21          $41 $     $8.6350 \cdot 10^{-7} $     $3.6216 \cdot 10^{-5} $    
22          $42 $     $8.6340 \cdot 10^{-7} $     $3.7095 \cdot 10^{-5} $    
23          $43 $     $8.6580 \cdot 10^{-7} $     $3.7864 \cdot 10^{-5} $    
24   $19\log 10 $     $8.6471 \cdot 10^{-7} $     $3.8044 \cdot 10^{-5} $    
25          $44 $     $7.8629 \cdot 10^{-7} $     $3.5371 \cdot 10^{-5} $    
26          $45 $     $5.0936 \cdot 10^{-7} $     $2.3423 \cdot 10^{-5} $    
27          $46 $     $3.3115 \cdot 10^{-7} $     $1.5559 \cdot 10^{-5} $    
28          $47 $     $2.1419 \cdot 10^{-7} $     $1.0278 \cdot 10^{-5} $    
29          $48 $     $1.3860 \cdot 10^{-7} $     $6.7893 \cdot 10^{-6} $    
..             ...                         ...                         ...   
217       $7100 $    $4.6632 \cdot 10^{-13} $     $3.3575 \cdot 10^{-9} $    
218       $7200 $    $3.8379 \cdot 10^{-13} $     $2.8017 \cdot 10^{-9} $    
219       $7300 $    $3.1583 \cdot 10^{-13} $     $2.3372 \cdot 10^{-9} $    
220       $7400 $    $2.5987 \cdot 10^{-13} $     $1.9491 \cdot 10^{-9} $    
221       $7500 $    $2.1380 \cdot 10^{-13} $     $1.6249 \cdot 10^{-9} $    
222       $7600 $    $1.7587 \cdot 10^{-13} $     $1.3542 \cdot 10^{-9} $    
223       $7700 $    $1.4466 \cdot 10^{-13} $     $1.1284 \cdot 10^{-9} $    
224       $7800 $    $1.1897 \cdot 10^{-13} $    $9.3984 \cdot 10^{-10} $    
225       $7900 $    $9.7830 \cdot 10^{-14} $    $7.8264 \cdot 10^{-10} $    
226       $8000 $    $8.0442 \cdot 10^{-14} $    $6.5158 \cdot 10^{-10} $    
227       $8100 $    $6.6138 \cdot 10^{-14} $    $5.4233 \cdot 10^{-10} $    
228       $8200 $    $5.4372 \cdot 10^{-14} $    $4.5129 \cdot 10^{-10} $    
229       $8300 $    $4.4695 \cdot 10^{-14} $    $3.7544 \cdot 10^{-10} $    
230       $8400 $    $3.6738 \cdot 10^{-14} $    $3.1228 \cdot 10^{-10} $    
231       $8500 $    $3.0196 \cdot 10^{-14} $    $2.5969 \cdot 10^{-10} $    
232       $8600 $    $2.4817 \cdot 10^{-14} $    $2.1591 \cdot 10^{-10} $    
233       $8700 $    $2.0395 \cdot 10^{-14} $    $1.7948 \cdot 10^{-10} $    
234       $8800 $    $1.6760 \cdot 10^{-14} $    $1.4917 \cdot 10^{-10} $    
235       $8900 $    $1.3772 \cdot 10^{-14} $    $1.2395 \cdot 10^{-10} $    
236       $9000 $    $1.1317 \cdot 

### Table 11: Sharper bounds for $\theta(x)$: $x$ in a middle range ($e^{20} \le x \le e^{13900}$)

Supplement for Table 10. All $\mathcal{B}_{j,k}$ values are valid for $x \in [e^{b_j},e^{13900}]$.

### Table 12: Lower bound for first values of $x \in [e^{J_0},10^{19}]$

Values of $\mathcal{C}_{b,k}$ in $\theta(x) > x - \frac{\mathcal{C}_{b,k} x}{(\log x)^k}$ calculated using the method described in Corollary 15.1, Section 4.4.
Each $\mathcal{C}_{b,k}$ is valid for $e^b \le x \le 10^{19} \simeq e^{43.749}$.

$$\begin{array}{cccccccc} 
 & b & \mathcal{C}_{b,1} & \mathcal{C}_{b,2} & \mathcal{C}_{b,3} & \mathcal{C}_{b,4} & \mathcal{C}_{b,5} & \\ 
 \hline
 &&&&&&& \\[-1em]
\rlap{\text{Calculated using } c=-0.8, C=0.81, \text{ each value valid up to } 5\cdot10^{10}\simeq e^{24.635}.}\\ \hline
&&&&&&& \\[-1em]
&  20  &  1.68440 \cdot 10^{-3}  &  3.36880 \cdot 10^{-2}  &  6.73750 \cdot 10^{-1}  &  1.34750 \cdot 10^{1}  &  2.69500 \cdot 10^{2 }  & \\
&  21  &  1.06840 \cdot 10^{-3}  &  2.24350 \cdot 10^{-2}  &  4.71140 \cdot 10^{-1}  &  9.89390 \cdot 10^{0}  &  2.07780 \cdot 10^{2 }  & \\
&  22  &  6.76540 \cdot 10^{-4}  &  1.48840 \cdot 10^{-2}  &  3.27450 \cdot 10^{-1}  &  7.20380 \cdot 10^{0}  &  1.58490 \cdot 10^{2 }  & \\
&  23  &  4.27800 \cdot 10^{-4}  &  9.83920 \cdot 10^{-3}  &  2.26310 \cdot 10^{-1}  &  5.20500 \cdot 10^{0}  &  1.19720 \cdot 10^{2 }  & \\
&  24  &  2.70120 \cdot 10^{-4}  &  6.48290 \cdot 10^{-3}  &  1.55590 \cdot 10^{-1}  &  3.73410 \cdot 10^{0}  &  8.96190 \cdot 10^{1 }  & \\
\hline
&&&&&&& \\[-1em]
\rlap{\text{Calculated using } c=-0.88, C=0.86, \text{ each value valid up to } 32\cdot10^{12}\simeq e^{31.097}.}\\ \hline
&&&&&&& \\[-1em]
&  \log(5\cdot 10^{10})  &  2.01560 \cdot 10^{-4}  &  4.96540 \cdot 10^{-3}  &  1.22330 \cdot 10^{-1}  &  3.01350 \cdot 10^{0}  &  7.42380 \cdot 10^{1 }  & \\
&  25  &  1.70330 \cdot 10^{-4} & 4.25830 \cdot 10^{-3}  &  1.06460 \cdot 10^{-1}  &  2.66140 \cdot 10^{0}  &  6.65350 \cdot 10^{1 }  & \\
&  26  &  1.10220 \cdot 10^{-4}  &  2.86560 \cdot 10^{-3}  &  7.45050 \cdot 10^{-2}  &  1.93720 \cdot 10^{0}  &  5.03650 \cdot 10^{1 }  & \\
&  27  &  6.93270 \cdot 10^{-5}  &  1.87190 \cdot 10^{-3}  &  5.05400 \cdot 10^{-2}  &  1.36460 \cdot 10^{0}  &  3.68430 \cdot 10^{1 }  & \\
&  28  &  4.35580 \cdot 10^{-5}  &  1.21970 \cdot 10^{-3}  &  3.41500 \cdot 10^{-2}  &  9.56180 \cdot 10^{-1}  &  2.67730 \cdot 10^{1 }  & \\
&  29  &  2.73380 \cdot 10^{-5}  &  7.92780 \cdot 10^{-4}  &  2.29910 \cdot 10^{-2}  &  6.66730 \cdot 10^{-1}  &  1.93360 \cdot 10^{1 }  & \\
&  30  &  1.71400 \cdot 10^{-5}  &  5.14180 \cdot 10^{-4}  &  1.54260 \cdot 10^{-2}  &  4.62760 \cdot 10^{-1}  &  1.38830 \cdot 10^{1 }  & \\
&  31  &  1.07350 \cdot 10^{-5}  &  3.32790 \cdot 10^{-4}  &  1.03170 \cdot 10^{-2}  &  3.19810 \cdot 10^{-1}  &  9.91400 \cdot 10^{0 }  & \\
\hline
&&&&&&& \\[-1em]
\rlap{\text{Calculated using } -c=C=0.94, \text{ each value valid up to } 10^{19}\simeq e^{43.749}.}\\ \hline
&&&&&&& \\[-1em]
&  \log(3.2\cdot10^{13})  &  1.02600 \cdot 10^{-5}  &  3.19040 \cdot 10^{-4}  &  9.92090 \cdot 10^{-3}  &  3.08510 \cdot 10^{-1}  &  9.59360 \cdot 10^{0 }  & \\
&  32  &  6.71750 \cdot 10^{-6}  &  2.14960 \cdot 10^{-4}  &  6.87870 \cdot 10^{-3}  &  2.20120 \cdot 10^{-1}  &  7.04380 \cdot 10^{0 }  & \\
&  33  &  4.38000 \cdot 10^{-6}  &  1.44540 \cdot 10^{-4}  &  4.76990 \cdot 10^{-3}  &  1.57410 \cdot 10^{-1}  &  5.19440 \cdot 10^{0 }  & \\
&  34  &  2.73610 \cdot 10^{-6}  &  9.30270 \cdot 10^{-5}  &  3.16300 \cdot 10^{-3}  &  1.07540 \cdot 10^{-1}  &  3.65640 \cdot 10^{0 }  & \\
&  35  &  1.70780 \cdot 10^{-6}  &  5.97730 \cdot 10^{-5}  &  2.09210 \cdot 10^{-3}  &  7.32220 \cdot 10^{-2}  &  2.56280 \cdot 10^{0 }  & \\
&  36  &  1.06520 \cdot 10^{-6}  &  3.83460 \cdot 10^{-5}  &  1.38050 \cdot 10^{-3}  &  4.96960 \cdot 10^{-2}  &  1.78910 \cdot 10^{0 }  & \\
&  37  &  6.63850 \cdot 10^{-7}  &  2.45630 \cdot 10^{-5}  &  9.08810 \cdot 10^{-4}  &  3.36260 \cdot 10^{-2}  &  1.24420 \cdot 10^{0 }  & \\
&  38  &  4.13450 \cdot 10^{-7}  &  1.57120 \cdot 10^{-5}  &  5.97020 \cdot 10^{-4}  &  2.26870 \cdot 10^{-2}  &  8.62100 \cdot 10^{-1 }  & \\
&  39  &  2.57330 \cdot 10^{-7}  &  1.00360 \cdot 10^{-5}  &  3.91400 \cdot 10^{-4}  &  1.52650 \cdot 10^{-2}  &  5.95320 \cdot 10^{-1 }  & \\
&  40  &  1.60060 \cdot 10^{-7}  &  6.40240 \cdot 10^{-6}  &  2.56100 \cdot 10^{-4}  &  1.02440 \cdot 10^{-2}  &  4.09750 \cdot 10^{-1 }  & \\
&  41  &  9.94970 \cdot 10^{-8}  &  4.07940 \cdot 10^{-6}  &  1.67260 \cdot 10^{-4}  &  6.85740 \cdot 10^{-3}  &  2.81160 \cdot 10^{-1 }  & \\
&  42  &  6.18140 \cdot 10^{-8}  &  2.59620 \cdot 10^{-6}  &  1.09040 \cdot 10^{-4}  &  4.57970 \cdot 10^{-3}  &  1.92350 \cdot 10^{-1 }  & \\
&  43  &  3.83820 \cdot 10^{-8}  &  1.65050 \cdot 10^{-6}  &  7.09680 \cdot 10^{-5}  &  3.05170 \cdot 10^{-3}  &  1.31220 \cdot 10^{-1 }  & \\
\hline
\end{array} $$

### Table 13: Lower bound for $x \le 10^{19}$ (Numerical Verification)

Values for $\mathcal{D}_k(a,b)$ in 
\\
 $\mathcal{D}_k(a,b) = \max_{n_0 \le n \le n_1} \left( \frac{(\log p_{n})^k \cdot (p_{n} - \theta(p_{n-1}))}{p_{n}} \right)$
\\
calculated using the method described in Lemma 16, Section 4.5. We give values for selected ranges between $1$ and $7.0 \cdot 10^{11}$.

### Table 14: Values for Theorem 2 for $k = 0$.

Values of $m_0$ in $\theta(x) \ge x(1-m_0)$ and values of $M_0$ in $\theta(x) \le x(1+M_0)$ from Theorem 2,
using the methods shown in Lemmas 18 and 17 respectively. Each value is valid for $x \ge \log X_0 = \log X_1$.

### Table 15: Values for Theorem 2 for $k \in \{1,2,3,4,5\}$.

Values of $m_k = M_k$ in $ |\theta(x) - x| < \frac{m_k x}{(\log x)^k}$ as in Theorem 2. 
Combined theoretical results from Table 8, 11, and 12. 
Each value is valid for $x \ge X_0 = X_1$. Note that for $x \le 10^{19}$, we have $M_k = 0$, and for $x \le e^{27}$, we have numerical results 
in Table 13.